OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -0,0 +1,457 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using ICSharpCode.SharpZipLib.Zip.Compression;
+using ICSharpCode.SharpZipLib.Zip.Compression.Streams;
+namespace OpenRA.FileSystem
+{
+	public class InstallShieldCABExtractor : IDisposable, IFolder
+	{
+		const uint FileSplit         = 0x1;
+		const uint FileObfuscated    = 0x2;
+		const uint FileCompressed    = 0x4;
+		const uint FileInvalid       = 0x8;
+
+		const uint LinkPrev          = 0x1;
+		const uint LinkNext          = 0x2;
+		const uint MaxFileGroupCount = 71;
+#region Nested Structs
+		struct FileGroup
+		{
+			public readonly string Name;
+			public readonly uint FirstFile;
+			public readonly uint LastFile;
+			public FileGroup(BinaryReader reader, long offset)
+			{
+				var nameOffset = reader.ReadUInt32();
+				/*   unknown  */ reader.ReadBytes(0x12);
+				FirstFile      = reader.ReadUInt32();
+				LastFile       = reader.ReadUInt32();
+
+				reader.BaseStream.Seek(offset + (long)nameOffset, SeekOrigin.Begin);
+				Name = reader.BaseStream.ReadASCIIZ();
+			}
+		}
+
+		struct VolumeHeader
+		{
+			public uint DataOffset;
+			public uint DataOffsetHigh;
+			public uint FirstFileIndex;
+			public uint LastFileIndex;
+
+			public uint FirstFileOffset;
+			public uint FirstFileOffsetHigh;
+			public uint FirstFileSizeExpanded;
+			public uint FirstFileSizeExpandedHigh;
+
+			public uint FirstFileSizeCompressed;
+			public uint FirstFileSizeCompressedHigh;
+			public uint LastFileOffset;
+			public uint LastFileOffsetHigh;
+
+			public uint LastFileSizeExpanded;
+			public uint LastFileSizeExpandedHigh;
+			public uint LastFileSizeCompressed;
+			public uint LastFileSizeCompressedHigh;
+
+			public VolumeHeader(BinaryReader reader)
```

Comment:
Could you please replace the `BinaryReader` with direct `ReadUInt32()` etc calls on the streams themselves?